[libmariadb:arm64-windows] build failure

**Host Environment**
- OS: Windows 10
- Compiler: MSVC v142

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install libmariadb:arm64-windows`

**Failure logs** 
```
[92/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_string.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_string.c
[93/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_sha1.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_sha1.c
[94/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\mariadb_stmt.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\mariadb_stmt.c
[95/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_stmt_codec.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_stmt_codec.c
[96/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\win32_errmsg.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\win32_errmsg.c
[97/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_dtoa.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_dtoa.c
[98/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DLIBMARIADB -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -Dma_getopt_EXPORTS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\unittest\mytap -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\unittest\libmariadb /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Founittest\libmariadb\CMakeFiles\ma_getopt.dir\ma_getopt.c.obj /Fdunittest\libmariadb\CMakeFiles\ma_getopt.dir\ /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\unittest\libmariadb\ma_getopt.c
[99/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_loaddata.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_loaddata.c
[100/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_client_plugin.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c libmariadb\ma_client_plugin.c
[101/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\mariadb_dyncol.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\mariadb_dyncol.c
[102/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_io.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_io.c
[103/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\mariadb_rpl.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\mariadb_rpl.c
[104/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\ma_context.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\ma_context.c
[105/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\mariadb_async.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\mariadb_async.c
[106/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DSIZEOF_CHARP=8 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\win-iconv /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -D HAVE_COMPRESS -D LIBMARIADB -D THREAD /showIncludes /Folibmariadb\CMakeFiles\mariadbclient.dir\secure\openssl.c.obj /Fdlibmariadb\CMakeFiles\mariadbclient.dir\mariadbclient.pdb /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\libmariadb\secure\openssl.c
[107/112] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\cl.exe   -DHAVE_DLOPEN -DHAVE_OPENSSL -DHAVE_TLS -DMARIADB_MACHINE_TYPE=\"AMD64\" -DMARIADB_SYSTEM_TYPE=\"Windows\" -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS -Dcctap_EXPORTS -IC:\vcpkg\installed\arm64-windows\include -Iinclude -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\auth -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\include -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\plugins\pvio -IC:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\zlib /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /wd4996 /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Founittest\mytap\CMakeFiles\cctap.dir\tap.c.obj /Fdunittest\mytap\CMakeFiles\cctap.dir\ /FS -c C:\vcpkg\buildtrees\libmariadb\src\8b17993a67-f2c4108afe.clean\unittest\mytap\tap.c
[108/112] cmd.exe /C "cd . && C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\lib.exe  /machine:ARM64 /nologo /machine:x64 /out:libmariadb\mariadbclient.lib libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\auth\my_auth.c.obj libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\auth\old_password.c.obj libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\pvio\pvio_socket.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_array.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_charset.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_hash.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_net.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_charset.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_time.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_default.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_errmsg.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_lib.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_list.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_pvio.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_tls.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_alloc.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_compress.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_init.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_password.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_ll2str.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_sha1.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_stmt.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_loaddata.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_stmt_codec.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_string.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_dtoa.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_rpl.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_client_plugin.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_io.c.obj libmariadb\CMakeFiles\mariadbclient.dir\secure\openssl.c.obj libmariadb\CMakeFiles\mariadbclient.dir\__\win-iconv\win_iconv.c.obj libmariadb\CMakeFiles\mariadbclient.dir\win32_errmsg.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_dyncol.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_async.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_context.c.obj  && cd ."
FAILED: libmariadb/mariadbclient.lib 
cmd.exe /C "cd . && C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\lib.exe  /machine:ARM64 /nologo /machine:x64 /out:libmariadb\mariadbclient.lib libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\auth\my_auth.c.obj libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\auth\old_password.c.obj libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\pvio\pvio_socket.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_array.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_charset.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_hash.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_net.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_charset.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_time.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_default.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_errmsg.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_lib.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_list.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_pvio.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_tls.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_alloc.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_compress.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_init.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_password.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_ll2str.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_sha1.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_stmt.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_loaddata.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_stmt_codec.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_string.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_dtoa.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_rpl.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_client_plugin.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_io.c.obj libmariadb\CMakeFiles\mariadbclient.dir\secure\openssl.c.obj libmariadb\CMakeFiles\mariadbclient.dir\__\win-iconv\win_iconv.c.obj libmariadb\CMakeFiles\mariadbclient.dir\win32_errmsg.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_dyncol.c.obj libmariadb\CMakeFiles\mariadbclient.dir\mariadb_async.c.obj libmariadb\CMakeFiles\mariadbclient.dir\ma_context.c.obj  && cd ."
libmariadb\CMakeFiles\mariadbclient.dir\__\plugins\auth\my_auth.c.obj : fatal error LNK1112: module machine type 'ARM64' conflicts with target machine type 'x64'
[109/112] cmd.exe /C "cd . && C:\vcpkg\downloads\tools\cmake-3.19.2-windows\cmake-3.19.2-win32-x86\bin\cmake.exe -E vs_link_dll --intdir=unittest\libmariadb\CMakeFiles\ma_getopt.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\link.exe  unittest\libmariadb\CMakeFiles\ma_getopt.dir\ma_getopt.c.obj  /out:unittest\libmariadb\ma_getopt.dll /implib:unittest\libmariadb\ma_getopt.lib /pdb:unittest\libmariadb\ma_getopt.pdb /dll /version:0.0 /machine:ARM64 /nologo    /debug /INCREMENTAL  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
[110/112] cmd.exe /C "cd . && C:\vcpkg\downloads\tools\cmake-3.19.2-windows\cmake-3.19.2-win32-x86\bin\cmake.exe -E vs_link_dll --intdir=unittest\mytap\CMakeFiles\cctap.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.293\bin\Hostx64\arm64\link.exe  unittest\mytap\CMakeFiles\cctap.dir\tap.c.obj  /out:unittest\mytap\cctap.dll /implib:unittest\mytap\cctap.lib /pdb:unittest\mytap\cctap.pdb /dll /version:0.0 /machine:ARM64 /nologo    /debug /INCREMENTAL  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
ninja: build stopped: subcommand failed.
```

**Additional context**
It is caused by `libmariadb/CMakeLists.txt` line 415
```
IF(CMAKE_SIZEOF_VOID_P EQUAL 8 AND MSVC)
  SET_TARGET_PROPERTIES(mariadbclient PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
ENDIF()
```
The devs hardcoded "/machine:x64".

--- a/port_versions/baseline.json
+++ b/port_versions/baseline.json
@@ -3074,7 +3074,7 @@
     },
     "libmariadb": {
       "baseline": "3.1.10",
-      "port-version": 4
+      "port-version": 5
     },
     "libmaxminddb": {
       "baseline": "1.4.3",

--- a/port_versions/l-/libmariadb.json
+++ b/port_versions/l-/libmariadb.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "674b0c8bcc3ee3276a0bce8afd72b99c5013f4a0",
+      "version-string": "3.1.10",
+      "port-version": 5
+    },
+    {
       "git-tree": "a5144f1b965487813bfd011e1d7ca18877d2ec69",
       "version-string": "3.1.10",
       "port-version": 4

--- a/ports/libmariadb/CONTROL
+++ b/ports/libmariadb/CONTROL
@@ -1,6 +1,6 @@
 Source: libmariadb
 Version: 3.1.10
-Port-Version: 4
+Port-Version: 5
 Homepage: https://github.com/MariaDB/mariadb-connector-c
 Description: MariaDB Connector/C is used to connect C/C++ applications to MariaDB and MySQL databases
 Default-Features: zlib, openssl

--- a/ports/libmariadb/arm64.patch
+++ b/ports/libmariadb/arm64.patch
@@ -1,0 +1,15 @@
+diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
+index 640a35bea..6c8d93250 100755
+--- a/libmariadb/CMakeLists.txt
++++ b/libmariadb/CMakeLists.txt
+@@ -410,10 +410,6 @@ TARGET_LINK_LIBRARIES(libmariadb LINK_PRIVATE ${SYSTEM_LIBS})
+
+ SIGN_TARGET(libmariadb)
+
+-IF(CMAKE_SIZEOF_VOID_P EQUAL 8 AND MSVC)
+-  SET_TARGET_PROPERTIES(mariadbclient PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
+-ENDIF()
+-
+ IF(CMAKE_SYSTEM_NAME MATCHES "Linux" OR
+    CMAKE_SYSTEM_NAME MATCHES "kFreeBSD" OR
+    CMAKE_SYSTEM_NAME MATCHES "GNU")

--- a/ports/libmariadb/portfile.cmake
+++ b/ports/libmariadb/portfile.cmake
@@ -9,6 +9,7 @@ vcpkg_from_github(
     SHA512 3e154f5dc4b5051607c7ebc0691a50c0699d60e4414660cf8f65689081ff78ef6b135667761ba8ac4163b469a3b55158c6b48c6fc0a0cc09381452aad157e4ad
     HEAD_REF 3.1
     PATCHES
+        arm64.patch
         md.patch
         disable-test-build.patch
         fix-InstallPath.patch

